vpr/src/place/net_cost_handler.cpp

```diff
@@ -54,6 +54,9 @@ enum class NetUpdateState {
 
 const int MAX_FANOUT_CROSSING_COUNT = 50;
 
+double cong_matrix[400][400];
```

Comment:
Needs comments on what these are, what the overall congestion modeling approach is.
The approach seems to be very similar to a placement cost function that I wrote in my PhD and called NON_LINEAR_CONGESTION.  At the time I measured that code it had little gain in routability for architectures with all channels the same width (in various chip regions) and the CPU time was high.  Not sure if the trade-off is different with this code, but adding QoR data on some larger circuits would be important to check.

Reply:
Also, hard-coding 400 is not good.